### DISCUSSION TITLE
Formatação de Data

### DISCUSSION THREAD

**Discussion by caioalbuka** [frontendbr/forum#2355] [Perguntas e Respostas]
Boa tarde, estou iniciando os estudos de Front-End e criando uma aplicação.
No momento tenho que formatar a data que vem do Mock para o formato de leitura do 'Calendar' da Biblioteca "react-multi-date-picker".

Estou travado nessa parte.

coreCalendary.ts

```js
import moment from "moment";
import api from "./apiCalendary";

interface CalendaryType {
  id: number,
  date: string,
  item: any
}

export const getCalendary = async () => {
  console.log('entrou')
  const result = await api.get('data-calendary')
  const resultEnd: string[] = []


  result.data.map((item: any) => {
    moment(item.date).format('DD/MM/YYYY')
    const formattedDate = moment(item.date).format('DD/MM/YYYY')
    console.log(
      'item', item.date,
      'format', formattedDate
    )
    resultEnd.push(formattedDate)
  })
  console.log('resultEnd', resultEnd)
  return resultEnd;

};
```

Não sei direito como se portar em foruns, se alguem puder me ajudar, caso precise de mais explicaçoes, toda ajuda é bem-vinda

**Comment by felquis** [comment]
Olá, bem vindo. Quando for compartilhar um pedaço de código, coloque dentro do block de código do github.

\```js
// código aqui
\```

Isso irá facilitar as pessoas te ajudarem <3

**Reply by caioalbuka** [reply]
Editei a msg, ajudou a me ajudarem?

**Comment by caioalbuka** [comment]
Oque esta acontecendo é que nao esta convertendo o timestamp para o tempo correto passado.
aparece sempre a mesma data no console.log ```` item 1663674065 format 20/01/1970````

**Comment by caioalbuka** [comment]
Consegui resolver, Alterei a formatação fazendo um novo json na Mock, com as datas que eu queria e não datas aleatórias.

**Comment by didiraja** [comment]
Para formatação de datas, recomendo o [date-fns](date-fns.org/) que é muito mais leve e acho a [documentação](https://date-fns.org/v2.29.2/docs/format) bem simples também
